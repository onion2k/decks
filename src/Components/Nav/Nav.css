
h1 {
    font-size: 30px; 
    font-size: 5vw;
    margin: 0;
    padding: 0;
}

.titleControls {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 0;
    align-items: center;
}

.yt1210Controls {
    justify-self: flex-end;
    text-align: right;
    list-style: none;
}

.yt1210Controls li {
    float: left;
    margin-right: 15px;
}

.yt1210Controls label {
    display: block;
    margin-bottom: 8px;
}

