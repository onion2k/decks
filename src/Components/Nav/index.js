import React, { Component } from "react";
import { Link } from "react-router-dom";

class Nav extends Component {
  render() {
    return (
      <ul className="yt1210Controls">
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/about">About</Link>
        </li>
      </ul>
    );
  }
}

export default Nav;
