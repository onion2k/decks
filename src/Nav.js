import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Nav extends Component {

    render() {
        return (
            <ul className="yt1210Controls">
                <li><Link to='/'>Home</Link></li>
                <li><Link to='/settings'>Settings</Link></li>
            </ul>
        );
    }
}

export default Nav;