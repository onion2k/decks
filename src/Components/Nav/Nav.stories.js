import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import StoryRouter from 'storybook-react-router';
import { linkTo } from '@storybook/addon-links';

import Nav from './index.js';

storiesOf('Nav', module)
  .addDecorator(
    StoryRouter(
      {
        '/': linkTo('Linked stories', 'home'),
        '/articles/*': linkTo('Linked stories', 'article'),
      },
      { initialEntries: ['/about'] }
    )
  )
  .add('Basic', () => <Nav />);
