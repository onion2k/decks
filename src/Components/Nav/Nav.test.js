import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom'

import Nav from ".";

describe("Navigation", () => {
  const onClick = jest.fn();

  it("renders properly", () => {
    const nav = renderer
      .create(
        <MemoryRouter>
          <Nav />
        </MemoryRouter>
      )
      .toJSON();
    expect(nav).toMatchSnapshot();
  });
});
