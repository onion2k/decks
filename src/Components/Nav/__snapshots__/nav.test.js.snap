// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders properly 1`] = `
<ul
  className="yt1210Controls"
>
  <li>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </li>
</ul>
`;
